client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStream.java

```diff
@@ -41,22 +41,22 @@
     /**
      * Flushes and then closes the output stream.
      * Frees any resources associated with it.
-     *
      * @throws SegmentSealedException If the segment is closed for modifications.
      */
     @Override
     public abstract void close() throws SegmentSealedException;
 
     /**
      * Block on all writes that have not yet completed.
-     *
      * @throws SegmentSealedException If the segment is closed for modifications.
      */
     public abstract void flush() throws SegmentSealedException;
 
     /**
-     * Returns a collection of all the events that have been passed to write but have not yet been
-     * acknowledged as written. The iteration order in the collection is from oldest to newest.
+     * Change the state of SegmentOutputStream to sealed to prevent future writes and return the list of unackedEvents.
+     * Any new writes will throw a SegmentSealedException.
```

Comment:
Yes. I meant on write. `write()` should now never throw a SegmentSealedException so it can be removed from the interface and in the impl we can turn it into an illegalState or something as it should not occur.